### PR TITLE
Sanity Check 1

### PR DESCRIPTION
**Pull request by KNayak430**
Kishan Nayak
33824899

### CODE REVIEW

**Comment by KNayak430** [comment]
Implemented the dice rolling module, which allows user to input a string of the form "2d6+ ... + 1d8+4" - which means two 6-sided dice, .... , one 8-sided die, and a constant of 4. The output simulates the sum (or difference) of the dice rolls as well as any constants. Any number of dice types may be used. 

Also implemented the player cache. Allows the program to load players from a txt file and also save to the same txt file. Player information is stored in a map consisting of Player structs. Each Player struct holds all the necessary information for a player character.

File: dndtool/lib/dndtool/player_cache.ex [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            stealth:       0, survival:        0
+
+end
+
+defmodule Player_Cache do
+
+  def start_link() do
+    Agent.start_link(fn -> load_players() end, name: __MODULE__ )
+  end
+
```
Comment (pragdave) [inline dndtool/lib/dndtool/player_cache.ex:24]: This module seems a little complex for what it does. 

When you save a player, to convert the map into a list of tuples, then write them out with a ":" between the key and value.

Can't you do the same when you read it back?

File: dndtool/test/dndtool_test.exs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+defmodule DnDToolTest do
+  use ExUnit.Case
+  doctest DnDTool
+
+  test "test roll bounds" do
+
```
Comment (pragdave) [inline dndtool/test/dndtool_test.exs:6]: Does this actually test the bounds reliably?

If you had an off-by-one error in a d20, you'd only find an error once in every 20 runs of the tests...

**Comment by pragdave** [comment]
# Approved

Just a suggestion. When you find yourself dealing with special cases, such as handling the `name` field, stop for a second, and ask if perhaps there's a different way of looking at things. Sometimes when code gives you trouble, it's trying to tell you something...

**Pull request closed without merging**